src/generator/helpers.ts

```diff
@@ -287,6 +287,55 @@ export function hasSchemaResponse(op: Operation): boolean {
   return false;
 }
 
+// returns the response envelope type name
+export function getResponseEnvelopeName(op: Operation): string {
+  return op.language.go!.responseEnv.language.go!.name;
+}
+
+// returns the final response envelope type name for LROs
+export function getFinalResponseEnvelopeName(op: Operation): string {
+  if (!isLROOperation(op)) {
+    throw new Error(`getFinalResponseEnvelopeName() called for ${op.language.go!.name} which isn't an LRO`);
+  }
+  return op.language.go!.finalResponseEnv.language.go!.name;
+}
+
+// returns the result envelope for the operation or undefined if it doesn't return a model
+export function hasResultEnvelope(op: Operation): Property | undefined {
```

Comment:
Small nit: the `has` makes me think this should return a boolean. Up to you if you're using this result somewhere below. 

Also why do we have this when we have `getResponseEnvelope` below? Does that make this function redundant?

Reply:
Oh nevermind I see it's response envelope versus result envelope...I'll keep looking at how these are used.

Reply:
This did originally return a boolean but was since changed.  I was on the fence about switching "has" to "get" since "get" IMO implies the thing exists.  I don't feel super-strong about it though so if you think "get" makes more sense I can rename it.

Also, see the comment in transform.ts for `createResponseEnvelope()` as it explains more about the response/result envelopes.